### ISSUE TITLE
AMD 15.7 Drivers 

### ISSUE THREAD

**Issue by ahayling** [jackun/openencodevfw#26]
AMD 15.7 Driver possible Issue.

I just installed AMD's new 15.7 drivers (from 14.12 Omega drivers) to test out the new VSR feature AMD rolled out, everything worked just fine. But for some reason the VFW encoder is now broken as I can no longer view the videos (using the decoder) or having no encoding options for the VFW encoder in Dxtory or Sony Vegas 13.

I thought it was a just a user error on my end but I decided to bring this up, jackun. Any help would be appreciated.

**Comment by jackun** [comment]
Did a search in driver folder and it seems to lack the openvideo.dll (or OVencode.dll or what it was named). Either they finally removed it or I failed at search :P Guess this is time to start on AMF version.

**Comment by i2apier** [comment]
Same problem here. At first I didn't think it was 15.7 driver that cause the issue. The encoder just didn't appear in compressor list of MSI After Burner.

**Comment by jmautari** [comment]
OpenVideo has been deprecated a long time ago and AMD dropped it starting at Catalyst 15.7 WHQL, any attempt to create an OpenVideo encoding session will fail and debug output will show an error "Cannot get pointer to external function: clCreateVideoEncSessionAMD".

Unfortunately it doesn't seem to be possible to just ship some DLLs from older Catalyst drivers to restore OpenVideo functionality so AMD AMF must be used instead to access VCE encoding capabilities. Too bad AMD AMF has so much overhead making it very slow compared to OpenVideo approach.

**Comment by i2apier** [comment]
I guess I'll roll back to 15.6 at the moment. Will wait to see the result of using AMF instead. :/

**Comment by nxrighthere** [comment]
Some software developers already solve this issue. http://capturegamevideo.com/index.php?/topic/4098-playclaw-51/?p=31146

**Comment by SpyPower** [comment]
I cannot make it work even with 15.6 catalyst version. MSVC2013 already installed both 32 an 64 (windows 8.1 64bit). Any solutions? any working catalyst version. ill try now with 14.12

**Comment by jackun** [comment]
I don't have 15.6 unpacked but 15.5 beta should work. See if its folders contain `openvideo.dll`

**Comment by SpyPower** [comment]
I think the problem is on my end. The openvideo.dll is there. I have currently installed the 14.12, Sherlock.exe shows me that the openencodevfw is installed but Dxtory does not display it as an option. tried to remove all codecs manually, but still it persists. :$. I don't know how else I could check if its working. no player gives me the option to select this codec.

**Comment by i2apier** [comment]
I'm using 15.6 driver and it's working fine on MSI AB. Try uninstall the driver with DDU (http://www.guru3d.com/files-details/display-driver-uninstaller-download.html) then install again.

![2015-07-20_04-06-58](https://cloud.githubusercontent.com/assets/8344037/8767865/d80d1538-2e94-11e5-831b-99d29bb2ffd2.png)

**Comment by SpyPower** [comment]
Well, as it seems, me too with MSI AB can use the codec. While playing (lol) at 40~60FPS, I record on 1050p (16:10) @ 25FPS. Same results with lagarith lossless. What I saw while recording is that the GPU utilization stays the same around 11%, the GPU D3D drops from 20~22% to 11%. It seems strange since it should be the double usage than this, cause of the recording by GPU.
Is there any way to make it work with Dxtory or OBS?

**Comment by SamJessop** [comment]
So the only fix is to roll back the drivers? Will this codec ever work now?

**Comment by computertechie** [comment]
Definitely looking forward to any news about openencode, it's the only thing I can get to work well with recording Planetside (CPU crushing game as it is).

**Comment by Rawi666** [comment]
@jackun Are you going to write AMD AMF vfw codec?

**Comment by jackun** [comment]
[Yeah,](https://github.com/jackun/TestAMFVFW) but it is a bit slow on older VCE atleast. YMMV.

**Comment by computertechie** [comment]
@jackun I just saw that repo earlier today, will give it a try tonight. I also have a 290 though, so I'll likely have similar performance as what you are getting.

**Comment by SpyPower** [comment]
@jackun 7870 Hd here. should use the old drivers?

**Comment by jackun** [comment]
@SpyPower  If you need to use OVE then yes.

**Comment by Rawi666** [comment]
Hmmm 15.7.1 and HD7970 and Dxtory doesn't see testamf codec. VC++ 2013 redistributable installed + all the stuff from VS2015. Is there any debug log or sth to help you in development?

**Comment by SpyPower** [comment]
15.7 HD7870 Dxtory doesn't see the teastamf codec too. MSI afterburner too cannot see the codec. Sherlock the codec detective cannot find it too. o.O

**Comment by computertechie** [comment]
Probably be best to open an issue on that repo :)

**Comment by jackun** [comment]
@Rawi666  @SpyPower Unpack to some permanent folder and install from there. Check that in registry `HKEY_CURRENT_USER\Software\TestAMFVFW\InstallPath` points to the `Runtimes` folder. Otherwise check with [DW](http://www.dependencywalker.com/), only thing unresolved should be AMF related dlls (ignoring ms-api-.... things).

**Comment by Stormy92** [comment]
I'm using it right now on Windows 10 with CCC 15.7.1 and Dxtory Version 2.0.128 

Works fine so far. It didn't appear in my Dxtory first  but i fixed it by simply restarting my computer. The perfomance is aboput 30-40 FPS for 1080p. But thank you so much for this. I was using OpenEncodeVFW for the last months and to see it won't work with newer drivers was kinda depressing for me.
 

![ss 2015-07-31 at 05 36 29](https://cloud.githubusercontent.com/assets/13589994/9011003/d347e1a8-37aa-11e5-8de4-d9e6526331b7.png)

**Comment by jackun** [comment]
@Stormy92 Biggest offender is the `SubmitInput` call that is used to send the buffer/surface to encoder. It just sits there 6ms doing god knows what. Even RGB-to-NV12 conversion is done in OpenCL in some ~2.5ms so the buffer is already on the graphics card and the whole thing would give just a little over 60fps in 1080p. Sigh...

**Comment by Stormy92** [comment]
@jackun Btw. Windows 10 is able to record gameplay with the build in Xbox app and it's using the GPU encoder on AMD and Nvidia GPUs too afaik. It's called "Game DVR"

Idk if this is helpful in any way for you.

**Comment by RavingCraze** [comment]
Do I understand right, from all the discussion here, that this whole codec is not working anymore? With the latest AMD driver? Is that why its just not showing anywhere? That would be soooo not nice T^T

I mean I can have very great Recordings with the AMD Evolved Gaming APP but I just hate, that it only uses one Audio line. That's why I'd prefer DxTory over everything else, but I can't record with my CPU! -,- My PC is bad!

**Comment by jackun** [comment]
@RavingCraze Yes, it has been depracted for long time already. Try the [AMF version](https://github.com/jackun/TestAMFVFW) but keep in mind that there's some API calls overhead that may limit you to about 1080p@45fps with first iterations of VCE atleast. Tonga (R9 285) and newer (R9 Fury (X)) _should_ be faster but I have none to test with.

**Comment by RavingCraze** [comment]
@jackun yeah lets talk about 1080p with 45FPS when I can run Arma 3 with that! (xD just kidding, thanks for the warning! ^^)

Thanks I'm gonna test it out! I Only record with 30FPS anyway!

**Comment by Prehistoricman** [comment]
It seems slightly wrong that AMD pioneered the overhead-removing Mantle, and then remove the fast API. Whatever the reason, thanks for providing the AMF version. It's the only way I can render quickly in Vegas.
